community/index/src/main/java/org/neo4j/index/gbptree/ConsistencyChecker.java

```diff
@@ -105,6 +228,34 @@ else if ( node.isLeaf( cursor ) )
         return true;
     }
 
+    private void assertPointerGenMatchesGen( PageCursor cursor, long expectedGen )
+    {
+        long nodeGen = node.gen( cursor );
+        assert nodeGen <= expectedGen : "Expected node:" + cursor.getCurrentPageId() + " gen:" + nodeGen +
+                " to be ≤ pointer gen:" + expectedGen;
+    }
+
+    private void checkNewGenPointerGen( PageCursor cursor ) throws IOException
+    {
+        long newGen = node.newGen( cursor, stableGeneration, unstableGeneration );
+        if ( TreeNode.isNode( newGen ) )
+        {
+            System.err.println( "WARNING: we ended up on an old generation " + cursor.getCurrentPageId() +
```

Comment:
Hmm, perhaps serving as additional debug information. This could be enforced later in the development process instead. Although I can't recall it even happening.